### PR TITLE
[STK-153][POC] -  Add Shutter RPC button

### PR DESCRIPTION
**Pull request by ElRodrigote**
## Description
* Adds the initial logic to connect through Metamask

### CODE REVIEW

**Comment by Diogomartf** [comment]
**Design feedback**
sometimes it breaks (probably we need some css to fix breaking)

image

**Comment by Diogomartf** [comment]
there are a couple of issues with this PR:

- this should only appear if the user is on gnosis chain (because it only works there so no point on showing on other networks)
- after adding the networks it keeps showing the button (if I'm on shutterized gnosis network already I don't need to see the button)
- the network should appear on the networks list after adding it, otherwise I have to keep adding it right? 
- why do I need this? why should I care? we need something to add more info to it or just linking it to shutter website where they explain.

I think we need to solve this issues in order to ship this.

**Comment by Diogomartf** [comment]
they updated their logo, so we should update it, it's a shield now. (maybe ask them the svg) 

image

Probably use their blue color as bg on the button? Might look better.

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -242,6 +269,35 @@ function Header() {
...
+          params: [chainParams],
+        })
+        console.log('Network added and switched to:', chainParams.chainName)
+      } catch (addError) {
+        console.error('Failed to add the network:', addError)
+      }
+    } else {
+      console.error('MetaMask is not installed!')
+    }
+  }
```
Comment (Diogomartf) [inline src/components/Header/index.tsx:299]: does it updating only the RPC works?



```js
async function updateNetworkRPC() {
  const chainId = '0x64'
  const newRpcUrl = 'https://erpc.gnosis.shutter.network'

  if (window.ethereum && window.ethereum.request) {
    try {
      await window.ethereum.request({
        method: 'wallet_addEthereumChain',
        params: [{
          chainId: chainId,
          rpcUrls: [newRpcUrl],
        }],
      })
      console.log('RPC URL updated for chain ID:', chainId)
    } catch (error) {
      // If the error code indicates the chain already exists, try updating the RPC
      if (error.code === 4902) {
        try {
          await window.ethereum.request({
            method: 'wallet_updateEthereumChain',
            params: [{
              chainId: chainId,
              rpcUrls: [newRpcUrl],
            }],
          })
          console.log('RPC URL updated for existing chain ID:', chainId)
        } catch (updateError) {
          console.error('Failed to update the RPC URL:', updateError)
        }
      } else {
        console.error('Failed to add or update the network:', error)
      }
    }
  } else {
    console.error('MetaMask is not installed!')
  }
}
```

**Review by Diogomartf — APPROVED** [review]

**Review by berteotti — APPROVED** [review]

**Pull request merged**